include/pylon_camera/internal/impl/pylon_camera_base.hpp

```diff
@@ -341,6 +371,47 @@ bool PylonCameraImpl<CameraTrait>::grab(std::vector<uint8_t>& image)
     return true;
 }
 
+template <typename CameraTrait>
+bool PylonCameraImpl<CameraTrait>::grab(std::vector<uint8_t>& image, ros::Time& stamp)
+{
+    Pylon::CGrabResultPtr ptr_grab_result;
+    if ( !grab(ptr_grab_result) )
+    {
+        ROS_ERROR("Error: Grab was not successful");
+        return false;
+    }
+
+    const uint8_t *pImageBuffer = reinterpret_cast<uint8_t*>(ptr_grab_result->GetBuffer());
+    image.assign(pImageBuffer, pImageBuffer + img_size_byte_);
+
+    /* Get pointer to the chunkTimestamp via the chunk data node map.
+     * To avoid delay caused by repeated dynamic road map creation,
+     * we create a static pool of node maps once at startup in enableTimestampChunk() above,
+     * as explained in Grab_ChunkImage.cpp of pylon C++ Sample code.
+     */
+    GenApi::CIntegerPtr chunkTimestamp(ptr_grab_result->GetChunkDataNodeMap().GetNode( "ChunkTimestamp"));
+    if (GenApi::IsReadable(chunkTimestamp))
+    {
+        uint64_t chunckTimestampCopy = chunkTimestamp->GetValue();
+        /* chunckTimestamp is based on a counter that counts the number of clock ticks generated by the camera.
+         * The unit of each tick is 8ns. So, we need to multiply the variable by 8.
+         * Note that multiplied value will wrap around from 0 if chunckTimestamp >= 2^61,
+         * or roughly 26687 days after the camera setup. */
+        chunckTimestampCopy <<= 3;
```

Comment:
@briansmlee Thanks for this implementation. I have tested it and confirmed this works. I wanted to point out an issue with hardcoding the mapping from ticks to time elapsed. I understand most cameras with PTP enabled have a mapping of `1 tick = 1ns` (`1 tick = 8ns` valid only when when PTP is disabled) for instance:

![image](https://user-images.githubusercontent.com/8161543/91455110-b1b96880-e84f-11ea-8988-599ab937ae4e.png)

You can confirm if there are any cameras that have a `1 tick = 8ns` mapping when PTP is enabled by displaying all camera models at: https://docs.baslerweb.com/timestamp.

In any case, it might be a good idea to have this as a parameter in the config file like you have done for enabling timestamp fetching.

Thanks,
Neel